apps/gateway/test/champions_test.exs

```diff
@@ -424,6 +428,142 @@ defmodule Gateway.Test.Champions do
+        response_type: {:battle_result, _ = battle_result}
+      }
+
+      assert battle_result.result == "win"
+
+      # Get advanced user
+      SocketTester.get_user(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = advanced_user}}
+
+      # Check that the gold and gems afk rewards rates are now greater than the initial rewards
+      gold_currency_id = Currencies.get_currency_by_name!("Gold").id
+      gems_currency_id = Currencies.get_currency_by_name!("Gems").id
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gold_currency_id && rate.rate > 0
+             end)
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gems_currency_id && rate.rate > 0
+             end)
+
+      # Claim afk rewards
+      gold_before_claiming = Currencies.get_amount_of_currency_by_name(advanced_user.id, "Gold")
+      gems_before_claiming = Currencies.get_amount_of_currency_by_name(advanced_user.id, "Gems")
+
+      # Simulate waiting 2 seconds before claiming the rewards
+      seconds_to_wait = 2
+      {:ok, advanced_user_db} = Users.get_user(advanced_user.id)
+
+      {:ok, _} =
+        advanced_user_db
+        |> GameBackend.Users.User.changeset(%{
+          last_afk_reward_claim: DateTime.utc_now() |> DateTime.add(-seconds_to_wait, :second)
+        })
+        |> Repo.update()
+
+      SocketTester.claim_afk_rewards(socket_tester, advanced_user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = claimed_user}}
+
+      # Check that the user has received gold and gems.
+      # The amount should be greater than the initial amount, and be in the range of the expected amount considering the time waited.
+      # We add 10% to the time waited to account for the time it takes to process the message.
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gold_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gold"))
+               reward_rate = Enum.find(claimed_user.afk_reward_rates, &(&1.currency_id == gold_currency_id)).rate
+               expected_amount = trunc(gold_before_claiming + reward_rate * seconds_to_wait)
+               user_gold_currency.amount in expected_amount..trunc(expected_amount * 1.1)
+             end)
+
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gems_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gems"))
+               reward_rate = Enum.find(claimed_user.afk_reward_rates, &(&1.currency_id == gems_currency_id)).rate
+               expected_amount = trunc(gems_before_claiming + reward_rate * seconds_to_wait)
+               user_gems_currency.amount in expected_amount..trunc(expected_amount * 1.1)
+             end)
```

Comment:
I think we can replace this two `Enum.any ` with an `Enum.all` and check that every currency in the user has the proper increase instead of hardcoding "Gold" and "Gems"